Automattic/wp-calypso
# Reader - Fix search stream locale

Related to https://github.com/Automattic/loop/issues/182

I found there is a race condition when the `switchLocale` function is called.

This function is called in `setLocaleMiddleware` -> `setLocale()`.
The `switchLocale` tries to load the language file for a valid `localeSlug`. The actual `I18N.prototype.getLocaleSlug` will not be set until this file is loaded via async HTTP request.
This means when we set the;
```
const lang = i18n.getLocaleSlug();
```
It will only work if this language file request is finished.

The problem is that when a logged out session on the Reader search page is loaded, the search query API request is often made before the language file is loaded, which means the search query defaults to using the `en` language. This means the initial search request can potentially (depending on cache!) some English sites for a non-english Reader search page like https://wordpress.com/fr/read/search.

## Before 
Shows inconsistent and crappy default search behaviour on Reader

https://github.com/Automattic/wp-calypso/assets/5560595/69b7e217-b9a7-43c5-9188-ca55e1e4e929

## After (English locale)

https://github.com/Automattic/wp-calypso/assets/5560595/4f6df219-c395-42f6-9c17-a4d36ce1d796

## After (Non-English locale)

https://github.com/Automattic/wp-calypso/assets/5560595/b9fcf19f-4089-4492-8182-41d2171552fa


## Proposed Changes

* This adds a `localeSlug` param to the `requestPage` to allow us to explicitly set the `lang` param in the API HTTP request.

## Why are these changes being made?


* Fixes broken search behaviour on Reader

## Testing Instructions



* Sandbox `public-api.wordpress.com` 
* Apply patch from code-D152262
* Go to logged out session for `/es/read/search` and `/fr/read/search` and confirm search results load as you'd expect
* Log into WPCOM and confirm Reader search works as you'd expect

## Pre-merge Checklist
